pandas/core/indexes/timedeltas.py

```diff
@@ -157,40 +156,26 @@ def __new__(cls, data=None, unit=None, freq=None, start=None, end=None,
             data = to_timedelta(data, unit=unit, box=False)
 
         if is_scalar(data):
-            raise ValueError('TimedeltaIndex() must be called with a '
-                             'collection of some kind, {data} was passed'
-                             .format(data=repr(data)))
+            raise TypeError(dtl.scalar_data_error(data, cls))
 
         # convert if not already
         if getattr(data, 'dtype', None) != _TD_DTYPE:
             data = to_timedelta(data, unit=unit, box=False)
         elif copy:
             data = np.array(data, copy=True)
 
-        data = np.array(data, copy=False)
-        if data.dtype == np.object_:
-            data = array_to_timedelta64(data)
-        if data.dtype != _TD_DTYPE:
-            if is_timedelta64_dtype(data):
-                # non-nano unit
-                # TODO: watch out for overflows
-                data = data.astype(_TD_DTYPE)
-            else:
-                data = ensure_int64(data).view(_TD_DTYPE)
+        arr = TimedeltaArrayMixin(data, freq=freq)
```

Comment:
I don't think we should replace the handling of object dtype with `TimedeltaArray` constructor (this should not be able to handle object dtype eventually), but if needed that is fine to leave for a later PR that actually does the split / cleans up the TimedeltaArray constructor

Reply:
I'm on board with the "leave for later" part of this